pkg/crc/preflight/preflight_checks_tray_windows.go

```diff
@@ -19,130 +17,119 @@ import (
-		password = strings.ReplaceAll(password, "`", "``")
-	}
-	if strings.Contains(password, "$") {
-		password = strings.ReplaceAll(password, "$", "`$")
-	}
-	if strings.Contains(password, "\"") {
-		password = strings.ReplaceAll(password, "\"", "`\"")
+func fixTrayInstalled() error {
+	/* Start the tray process and copy the executable to start-up folder
+	 * "$USERPROFILE\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup"
+	 */
+	cmd := fmt.Sprintf(`New-Item -ItemType SymbolicLink -Path "%s" -Name "%s" -Value "%s"`,
+		constants.StartUpFolder,
+		constants.TrayShortcutName,
+		constants.TrayExecutablePath,
+	)
+	if _, _, err := powershell.ExecuteAsAdmin("Create symlink to tray in start-up folder", cmd); err != nil {
+		return fmt.Errorf("Error trying to create symlink to tray in start-up folder: %w", err)
 	}
-	if strings.Contains(password, "'") {
-		password = strings.ReplaceAll(password, "'", "`'")
+	cmd = fmt.Sprintf(`Start-Process -FilePath "%s"`, constants.TrayExecutablePath)
+	if _, _, err := powershell.Execute(cmd); err != nil {
+		return fmt.Errorf("Failed to start tray process: %w", err)
 	}
-	return password
+	return nil
 }
 
 func removeTray() error {
-	trayProcessName := constants.TrayExecutableName[:len(constants.TrayExecutableName)-4]
-
-	tempDir, err := ioutil.TempDir("", "crc")
-	if err != nil {
-		return err
+	var mErr errors.MultiError
+	if err := goos.Remove(filepath.Join(constants.StartUpFolder, constants.TrayShortcutName)); err != nil {
+		mErr.Collect(err)
+	}
+	if err := goos.Remove(constants.DaemonBatchFilePath); err != nil {
+		mErr.Collect(err)
+	}
+	if err := goos.Remove(constants.DaemonPSscriptPath); err != nil {
+		mErr.Collect(err)
 	}
-	defer func() {
-		_ = goos.RemoveAll(tempDir)
-	}()
 
-	psScriptContent := genTrayRemovalScript(
-		trayProcessName,
-		constants.TrayShortcutName,
-		constants.DaemonServiceName,
-		tempDir,
-	)
-	psFilePath := filepath.Join(tempDir, "trayRemoval.ps1")
+	logging.Info("Please run the following powershell command to finish the cleanup process.")
+	logging.Info("This will stop the CodeReady Containers daemon process: '(Get-Process crc).kill()'")
 
-	// write script content to temporary file
-	if err = writePsScriptContentToFile(psScriptContent, psFilePath); err != nil {
-		return err
+	if len(mErr.Errors) == 0 {
+		return nil
 	}
+	return mErr
+}
 
-	_, _, err = powershell.ExecuteAsAdmin("Uninstalling CodeReady Containers System Tray", psFilePath)
-	// wait for the script to finish executing
-	time.Sleep(2 * time.Second)
-	if err != nil {
-		return err
+func stopTray() error {
+	trayProcessName := constants.TrayExecutableName[:len(constants.TrayExecutableName)-4]
+	cmd := fmt.Sprintf(`Stop-Process -Name "%s"`, trayProcessName)
+	if _, _, err := powershell.Execute(cmd); err != nil {
+		return fmt.Errorf("Failed to kill tray process: %w", err)
 	}
 	return nil
 }
 
+func removeDaemon() error {
+	return goos.Remove(constants.DaemonBatchFilePath)
```

Comment:
You already remove this in removeTray()?

Reply:
Yup, messed up during the squash i think, will fix this